src/app.js

```diff
@@ -8,10 +8,12 @@ import hash from './hash'
 export default class App extends Component {
   constructor(props) {
     super(props)
+    const stateJson = localStorage.getItem('view-app-state')
+    const appState = JSON.parse(stateJson) || {}
     const { path } = hash.parse(location.hash)
     this.state = {
-      view: { path },
-      cards: []
+      view: appState.view || { path },
```

Comment:
The persisted state should not contain `view` information because the browser's history (back/forward) can keep track of that. So you can get rid of this `||` guard for `view`.